packages/aws-cdk-lib/aws-iam/lib/principals.ts

```diff
@@ -481,6 +481,9 @@ export class AccountPrincipal extends ArnPrincipal {
     if (!cdk.Token.isUnresolved(accountId) && typeof accountId !== 'string') {
       throw new Error('accountId should be of type string');
     }
+    if (!cdk.Token.isUnresolved(accountId) && !this.accountIdRegExp.test(accountId)) {
+      throw new Error('accountId should be composed of 12 digits');
+    }
```

Comment:
I wonder if there might be use cases where people are passing valid strings that aren't exactly 12 digits in length.

I'd feel more comfortable making this a warning, rather than an exception.